### ISSUE TITLE
Misleading "Not built with GPU enabled. Skip GPU library dlopen check." message when using GPU 2.0 nightly build

### ISSUE THREAD

**Issue by wdirons** [tensorflow/tensorflow#28940]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No 
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04 x86
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): pip install tf-nightly-gpu-2.0-preview==2.0.0.dev20190521
- TensorFlow version (use command below): ('v1.12.1-2420-g4fd6e533f6', '2.0.0-dev20190521')
- Python version:  2.7
- Bazel version (if compiling from source): N/A
- GCC/Compiler version (if compiling from source): N/A
- CUDA/cuDNN version: 10.0 / 7
- GPU model and memory: P100 16 GB

**Describe the current behavior**

TensorFlow compiled with GPU support prints the misleading message:
```
I tensorflow/stream_executor/platform/default/dlopen_checker.cc:62] Not built with GPU enabled. Skip GPU library dlopen check.
```

This confuses the user into believe GPU is not working (or at least not fully working), when it really is.

**Describe the expected behavior**

The message should not be printed when built with GPU support, or should be worded better.

**Code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate the problem.

```
pip install tf-nightly-gpu-2.0-preview==2.0.0.dev20190521
python
import tensorflow as tf
input_string = 'Hello, TensorFlow!'
hello = tf.constant(input_string)
```

**Other info / logs**
```
root@f2ce3e981b16:~# python
Python 2.7.15rc1 (default, Nov 12 2018, 14:31:15)
[GCC 7.3.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import tensorflow as tf
>>> input_string = 'Hello, TensorFlow!'
>>> hello = tf.constant(input_string)
2019-05-22 18:10:31.116739: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcuda.so.1
2019-05-22 18:10:31.936204: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1640] Found device 0 with properties:
name: Tesla P100-PCIE-16GB major: 6 minor: 0 memoryClockRate(GHz): 1.3285
pciBusID: 0000:06:00.0
2019-05-22 18:10:31.937373: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1640] Found device 1 with properties:
name: Tesla P100-PCIE-16GB major: 6 minor: 0 memoryClockRate(GHz): 1.3285
pciBusID: 0000:81:00.0
2019-05-22 18:10:31.937394: I tensorflow/stream_executor/platform/default/dlopen_checker.cc:62] Not built with GPU enabled. Skip GPU library dlopen check.
2019-05-22 18:10:31.942721: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1763] Adding visible gpu devices: 0, 1
2019-05-22 18:10:31.944135: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2019-05-22 18:10:32.163526: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x5611c578aac0 executing computations on platform CUDA. Devices:
2019-05-22 18:10:32.163568: I tensorflow/compiler/xla/service/service.cc:175]   StreamExecutor device (0): Tesla P100-PCIE-16GB, Compute Capability 6.0
2019-05-22 18:10:32.163580: I tensorflow/compiler/xla/service/service.cc:175]   StreamExecutor device (1): Tesla P100-PCIE-16GB, Compute Capability 6.0
2019-05-22 18:10:32.167321: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 2199905000 Hz
2019-05-22 18:10:32.171784: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x5611c5bc4df0 executing computations on platform Host. Devices:
2019-05-22 18:10:32.171814: I tensorflow/compiler/xla/service/service.cc:175]   StreamExecutor device (0): <undefined>, <undefined>
2019-05-22 18:10:32.173360: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1640] Found device 0 with properties:
name: Tesla P100-PCIE-16GB major: 6 minor: 0 memoryClockRate(GHz): 1.3285
pciBusID: 0000:06:00.0
2019-05-22 18:10:32.174698: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1640] Found device 1 with properties:
name: Tesla P100-PCIE-16GB major: 6 minor: 0 memoryClockRate(GHz): 1.3285
pciBusID: 0000:81:00.0
2019-05-22 18:10:32.174718: I tensorflow/stream_executor/platform/default/dlopen_checker.cc:62] Not built with GPU enabled. Skip GPU library dlopen check.
2019-05-22 18:10:32.179748: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1763] Adding visible gpu devices: 0, 1
2019-05-22 18:10:32.180155: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudart.so.10.0
2019-05-22 18:10:32.183049: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1181] Device interconnect StreamExecutor with strength 1 edge matrix:
2019-05-22 18:10:32.183069: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1187]      0 1
2019-05-22 18:10:32.183083: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1200] 0:   N N
2019-05-22 18:10:32.183093: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1200] 1:   N N
2019-05-22 18:10:32.189364: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1326] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 15216 MB memory) -> physical GPU (device: 0, name: Tesla P100-PCIE-16GB, pci bus id: 0000:06:00.0, compute capability: 6.0)
2019-05-22 18:10:32.191715: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1326] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:1 with 15216 MB memory) -> physical GPU (device: 1, name: Tesla P100-PCIE-16GB, pci bus id: 0000:81:00.0, compute capability: 6.0)
```

**Comment by wdirons** [comment]
@yifeif , this message was added in this commit:
https://github.com/tensorflow/tensorflow/commit/fcfb99bc3aa3110feae5233dbe4634bffa70dc59

Is the #if GOOGLE_CUDA variable not getting set correctly ?

**Comment by yifeif** [comment]
Thanks @wdirons! I also noticed this and it should be fixed in https://github.com/tensorflow/tensorflow/commit/eae5d0bc15bf8bdb7635a9692fb19d747352a445. 
Could you give the latest tf-nightly-gpu a try?

**Comment by wdirons** [comment]
I verified this fix with `pip install tf-nightly-gpu-2.0-preview==2.0.0.dev20190523`

Thank you @yifeif!

**Comment by victormaggie** [comment]
Hi wdirons. thanks for your solution. You save my life.. I can run cuda and cublas now.

by using 👍 
pip install tf-nightly-gpu-2.0-preview==2.0.0.dev20191002
!!
